### DISCUSSION TITLE
Power Panels

### DISCUSSION THREAD

**Discussion by JHarrison712** [netbox-community/netbox#12511] [Q&A] [answered]
We are setting up NetBox at our organization and we have a question about the Power Panel section.

We don't want to keep creating Power Panels for each site. Because after it goes to the Power Panel> Then the site gets to know what the feed they get. https://i.imgur.com/H76Czam.png 

This is my demo lab I have you can see here I have to create 2 sub panel that goes into the same location but different site name. https://i.imgur.com/vP8Wmaj.png

I would like to see the Power Panel not really Link to an site just say Panel A > Power Feed gets know what site its going to be connected too

**Comment by candlerb** [comment]
I'm afraid I don't understand what you're saying, and the blurred out images make it impossible to see.

You said: "I have to create 2 sub panel that goes into the same location but different site name". That doesn't make any sense. In Netbox, a Location by definition is a subdivision *within* a Site. It is (or should be) impossible to create any object which is linked to the same Location but in a different Site, because the Location itself is fixed to a specific Site.

You could however have two different locations, within two different sites, and those locations happen to have the same name (e.g. "Battery Room" in site A, and "Battery Room" in site B - these are different Locations)

Can you give the exact sequence of steps to reproduce what you are seeing? And also, say what version of Netbox you are using?

You also said: "I would like to see the Power Panel not really Link to an site just say Panel A > Power Feed gets know what site its going to be connected too".

The physical hierarchy within Netbox goes like this:

Region* > Site > Location* > Rack

("*" indicates both optional and recursive, e.g. Locations can contain child Locations)

For power, the model looks like this:
* A Power Panel must be linked to a Site, and optionally to a Location within that site.
* A Power Feed must be linked to a Power Panel, and optionally to a Rack

That is, a Power Panel represents a collection of power feeds, and gives the overall physical location of them all, down to the level of "Location" (which is typically a single room, a cage, or a row of racks).

There is no direct relationship between Power Feed and Site. The location of any Power Feed is implicitly the location of its Power Panel, but it can be linked more specifically to a specific Rack within that Location. This allows you to have one Power Panel in a room, but present the Power Feeds to individual racks.

Therefore, if you want to model one or more Power Feeds for a site, when you don't yet know where within that site they are going to be located, you must:
1. Create a Power Panel for that site, with Location unset
2. Create one or more Power Feeds, linked to that Power Panel, with Rack unset

**Reply by JHarrison712** [reply]
Here is a better screenshot https://i.imgur.com/OXWM22A.png

**Reply by candlerb** [reply]
Do you have three different locations called "Garage Room Closet", one in each of those three sites?

**Reply by JHarrison712** [reply]
I do https://i.imgur.com/9RzhnNU.png & https://i.imgur.com/OABet2s.png

**Reply by candlerb** [reply, marked as answer]
OK, then it's working as expected.

As I said, a Location is a subdivision of a Site. You can't have a Location without a Site, and a specific Location belongs to only one Site.

Pretty much everything physical in Netbox must be linked to a Site: for example, a device can *optionally* be in a rack, and *optionally* in a location, but *must* at least be in a Site.

The same applies to Power Panels. It applies indirectly to Power Feeds too (because a Power Feed must be contained within a Power Panel; and a Power Panel must be within a Site)

I hope I've explained this clearly.

You said before:

> I would like to see the Power Panel not really Link to an site just say Panel A > Power Feed gets know what site its going to be connected too

The sentence structure doesn't make sense to me, so can you rephrase?

The way a power feed knows what site it is in, is by linking it to a power panel in a site. And a power panel which isn't in *any* site isn't allowed in Netbox, in the same way as a device which isn't in any site isn't allowed.
